### DISCUSSION TITLE
Template event - add ability to overwrite template file path

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#12245] [Ideas] [answered]
Hi! Would it be possible to add? I need such functionality for my theming-related plugin.
I am talking abut events like `beforeRenderTemplate` or `beforeRenderPageTemplate` from `web\View` class. 

Making overwriting template file path possible would be really simple, just add `$template = $event->template;` to function and make `$template` variable being passed by reference, just like `$templateMode` and `$variables` are.

**Comment by brandonkelly** [comment, marked as answer]
Seems like this could lead to a lot of confusion.

Theming plugins can register their own custom [template roots](https://craftcms.com/docs/4.x/extend/template-roots.html) and allow developers to explicitly reference their provided templates.
